### ISSUE TITLE
Manual seeding from command line option

### ISSUE THREAD

**Issue by bilzard** [ultralytics/yolov5#6622]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar feature requests.

### Description

Currently, the initial random seed is fixed in `train.py`[1].
It is problematic when doing fine-tune: it produces exactly the same data as the pre-training (suppose we use the same training data). That is likely to cause overfitting.

## Possible Choice

1. seed from current timestamps -> can't reproduce the same result
2. seed from command line option
3. save random state as the checkpoint -> YOLOv5 doesn't save training state if one training is over[2]. So this implementation is inconsistent to the current YOLOv5 repo.

I think option 2 is better than others.
@glenn-jocher, what do you think? Any other options?

### Reference

- [1] https://github.com/ultralytics/yolov5/blob/master/train.py#L108
- [2] https://github.com/ultralytics/yolov5/blob/master/train.py#L427

### Use case

Seed by command line option:

```
train.py ... --seed 1234
```

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@bilzard yes good point. If we just want random or reproducible, maybe just --seed random and then use timestamp?

**Comment by bilzard** [comment]
@glenn-jocher Sorry, I don't understand the point. 

1. Does't just `--seed` by command line parameter works O.K? Why seeding by timestamp is needed?
2. I can't imagine how two type of seeding is possible at the same time. Can you explain more detail?

**Comment by glenn-jocher** [comment]
@bilzard I guess you'd want two options:

`--seed 0` (default, or pass any other int or float)
or
`--seed random` (string argument, use timestamp)

**Comment by bilzard** [comment]
What is the purpose of `--seed random`?
It won't be able to reproduce the result.
So I think the first option is better.

**Comment by glenn-jocher** [comment]
@bilzard sure, we can just a --seed integer argument, default 0
